src/lib/statsig.ts

```diff
@@ -1,20 +1,69 @@
-// import { StatsigClient } from '@statsig/js-client';
+import { StatSigFlags, StatsigConfigType } from '@/types/statsig';
+import { StatsigClient } from '@statsig/js-client';
+import { merge } from 'lodash';
 
-// let statsigClient: StatsigClient;
+import { log } from './telemetry';
 
-// export const initStatsig = async () => {
-//   if (statsigClient) return statsigClient;
-//   statsigClient = new StatsigClient(
-//     import.meta.env.VITE_STATSIG_CLIENT_KEY,
-//     {},
-//     {
-//       disableLogging: import.meta.env.VITE_DISABLE_STATSIG,
-//       disableStorage: import.meta.env.VITE_DISABLE_STATSIG,
-//     }
-//   );
-//   await statsigClient.initializeAsync();
-//   return statsigClient;
-// };
+let statsigClient: StatsigClient;
+let initPromise: Promise<StatsigClient> | null = null;
 
-// TODO: figure out how to init statsig async so we get configs on the first page load without breaking app state
-// https://linear.app/dydx/project/feature-experimentation-6853beb333d7/overview
+/**
+ * This method creates a promise and assigns it to the variable initPromise.
+ * If an initPromise has already been generated, it returns the existing promise.
+ * This causes all calls to this function to always return the statsigClient only after
+ * it has retrieved all values from statsig
+ *
+ * @returns initPromise: a promise that returns the statsig client only after it is initialized
+ */
+export const initStatsig = async () => {
+  if (initPromise) {
```

Comment:
yeah it's a little bit annoying. i spent some time thinking about this last night but the only other thing i coudl think of would be to put the entire app into a loading state while the promise was running, which i think is a cleaner but possibly bit trickier solution since we'd also need to put prevent all the providers from running as well 🤔.

FWIW i have a project to bolster our experimentation https://linear.app/dydx/project/feature-experimentation-6853beb333d7/issues so illl be investigating this later.